src/sas/sascalc/data_util/qsmearing.py

```diff
@@ -91,7 +119,17 @@ class PySmear(object):
     def __init__(self, resolution, model):
         self.model = model
         self.resolution = resolution
-        self.offset = numpy.searchsorted(self.resolution.q_calc, self.resolution.q[0])
+
+        if hasattr(self.resolution, 'data'):
+            if self.resolution.data.meta_data['loader'] == 'SESANS':  # Always True if file extension is '.ses'!
+                self.offset = 0
+            # This is default behaviour, for future resolution/transform functions this needs to be revisited.
+            else:
+                self.offset = numpy.searchsorted(self.resolution.q_calc, self.resolution.q[0])
+        else:
+            self.offset = numpy.searchsorted(self.resolution.q_calc, self.resolution.q[0])
+
+        # self.offset = numpy.searchsorted(self.resolution.q_calc, self.resolution.q[0])
```

Comment:
That line is not my code, only the metadata check above it is mine.

q_calc must envelop q, so q_calc[0] < q[0] and q_calc[-1] > q[-1].
In words: The Hankel transform integration q-space (q_calc) must envelop the sasmodel q-space (q)
So, setting them equal will break the transform.